compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -633,8 +633,9 @@ static void addLowerToLLVMPasses(OpPassManager &passManager) {
     passManager.addNestedPass<func::FuncOp>(
         createLLVMCPUEmitVectorizationRemarksPass());
   }
-  passManager.addPass(IREE::Util::createPromoteArithBF16ToF32Pass());
   passManager.addNestedPass<func::FuncOp>(createConvertLinalgToLoopsPass());
+  passManager.addPass(IREE::Util::createPromoteArithBF16ToF32Pass());
```

Comment:
Confused about these two passes. The first one is promoting to f32, and then the second one is emulating it after promotion?

Reply:
I should probably rename. One promotes all arithmetic/math operations from `bf16` to `f32`. The second replaces all vectorized loads / stores with `i16` equivalents to remove the `bf16` type from the LLVM backend. I had hoped that load/store operations of `bf16` would just translate to equivalent bit-lengths but unfortunately I was wrong.

Reply:
Do you mind renaming this? I didnt know i16 was also implicated here (but thats because I dont understand bf16 and its emulation)

Reply:
I renamed to better explain. The arith conversion handles the simulated arithmetic operations (now include `vector.reduce`) however the EmulateBF16 is focused on the buffer operations (loading, storing, subspan). Arguably it shouldn't be required as loading / storing bytes should not matter about the type being operated on, however LLVM apparently disagrees.

This fix also now puts the `cuda` pipeline down the same path.